### DISCUSSION TITLE
How to execute 2 BeforeSuite hooks in a given order?

### DISCUSSION THREAD

**Discussion by zbrea** [getgauge/gauge#2106] [General]
I want to run all the scenarios in a test suite without closing / restarting the app.

How I can start , in order, the Appium server,  then the AndroidDriver to launch the application with the __BeforeSuite__ hook ?

![image](https://user-images.githubusercontent.com/59898499/132998030-87caf011-7128-441c-93fd-367a900de764.png)

**Comment by sriv** [comment]
Rather than depending on the order of invocation of the hooks, you can use a single `BeforeSuite` hook that invokes these functions in order. That way you can also handle the exceptions within the hook.

**Comment by zbrea** [comment]
Good tip!   Thank you !
